### ISSUE TITLE
Improve storage of favicons

### ISSUE THREAD

**Issue by Emdek** [OtterBrowser/otter-browser#820]
Reliability of favicons storage (as part of global history) needs improvements.

**Comment by VSHY** [comment]
That global storage of favicons would be better to use for searches also. Now the favicons are stored in \searches\*.xml-files.

**Comment by Emdek** [comment]
@VSHY, nah, they use embedded icons for a reason, it's part of the standard (it's customizable using built-in editor) and won't randomly change if service will choose to change own icon (like Google did multiple times in the past).

**Comment by VSHY** [comment]
@Emdek,
 In my some Opera skin was the old icon. Time passed and I changed it to a new, because it did not correspond icon which located into the address bar, history, etc.
 Therefore, a static icon - not a good idea.
 It has two different logic, and each has a right to exist.

**Comment by Emdek** [comment]
@VSHY, this is why that standard allows to embed link to original resource so it can be updated from time to time. ;-)

**Comment by beastie1** [comment]
I wish favicons were displayed even when images are disabled globally.

**Comment by Emdek** [comment]
@beastie1, uhm, are they affected by that setting?
If yes then it would be another reason to use custom algorithm to fetch them in QtWebKit backend (QtWebEngine needs that anyway)...

**Comment by VSHY** [comment]
@Emdek
Will you add the button for updating of the icon in the certain search later? :)

**Comment by Emdek** [comment]
@VSHY, currently it can be set manually using search engine editor, that should be good enough in most cases.

**Comment by VSHY** [comment]
I found it) Thanks.

**Comment by cofi89** [comment]
Hi @Emdek,
Any status/plans/ETA on this?

**Comment by Emdek** [comment]
@cofi89, sure, I forgot to readd milestone.
There are some initial designs but nothing solid yet.
